pkg/pipeline/CdHandler.go

```diff
@@ -170,6 +176,63 @@ type ArgoPipelineStatusSyncEvent struct {
 	IsAppStoreApplication bool  `json:"isAppStoreApplication"`
 }
 
+type CmBlobStorageConfig struct {
+	//AWS credentials
+	CloudProvider               blob_storage.BlobStorageType `json:"BLOB_STORAGE_PROVIDER"`
+	S3AccessKey                 string                       `json:"BLOB_STORAGE_S3_ACCESS_KEY"`
+	S3Endpoint                  string                       `json:"BLOB_STORAGE_S3_ENDPOINT"`
+	S3EndpointInsecure          string                       `json:"BLOB_STORAGE_S3_ENDPOINT_INSECURE"`
+	S3BucketVersioned           string                       `json:"BLOB_STORAGE_S3_BUCKET_VERSIONED"`
+	CdDefaultBuildLogsBucket    string                       `json:"DEFAULT_BUILD_LOGS_BUCKET" `
+	CdDefaultCdLogsBucketRegion string                       `json:"DEFAULT_CD_LOGS_BUCKET_REGION" `
+	DefaultCacheBucket          string                       `json:"DEFAULT_CACHE_BUCKET"`
+	DefaultCacheBucketRegion    string                       `json:"DEFAULT_CACHE_BUCKET_REGION"`
+
+	//Azure credentials
+	AzureAccountName               string `json:"AZURE_ACCOUNT_NAME"`
+	AzureGatewayUrl                string `json:"AZURE_GATEWAY_URL"`
+	AzureGatewayConnectionInsecure string `json:"AZURE_GATEWAY_CONNECTION_INSECURE"`
+	AzureBlobContainerCiLog        string `json:"AZURE_BLOB_CONTAINER_CI_LOG"`
+	AzureBlobContainerCiCache      string `json:"AZURE_BLOB_CONTAINER_CI_CACHE"`
+}
+
+func (c CmBlobStorageConfig) PopulateWithK8sExtBlobCmData(cm map[string]string) (CmBlobStorageConfig, error) {
+	cmDataJson, err := json.Marshal(cm)
+	if err != nil {
+		fmt.Println("error marshalling external blob storage cm data to json:", err)
+		return c, err
```

Comment:
why is it returning `CmBlobStorageConfig` object ?

Reply:
resolved this as well, no need to return the CmBlobStorageConfig object, that will populate itself when method defined on pointer of that object